airbyte-integrations/connectors/destination-pinecone/destination_pinecone/document_processor.py

```diff
@@ -0,0 +1,121 @@
+
+    def __init__(self, config: ProcessingConfigModel, catalog: ConfiguredAirbyteCatalog, max_metadata_size: Optional[int] = None):
+        self.streams = {self._stream_identifier(stream.stream): stream for stream in catalog.streams}
+        self.max_metadata_size = max_metadata_size
+
+        self.splitter = RecursiveCharacterTextSplitter.from_tiktoken_encoder(
+            chunk_size=config.chunk_size, chunk_overlap=config.chunk_overlap
+        )
+        self.metadata_fields = config.meta_data_fields
+        self.text_fields = config.text_fields
+        self.logger = logging.getLogger("airbyte.document_processor")
+
+    def _stream_identifier(self, stream: Union[AirbyteStream, AirbyteRecordMessage]) -> str:
+        if isinstance(stream, AirbyteStream):
+            return stream.name if stream.namespace is None else f"{stream.namespace}_{stream.name}"
+        else:
+            return stream.stream if stream.namespace is None else f"{stream.namespace}_{stream.stream}"
+
+    def process(self, record: AirbyteRecordMessage) -> Tuple[List[Document], Optional[str]]:
+        """
+        Generate documents from records.
+        :param records: List of AirbyteRecordMessages
+        :return: Tuple of (List of document chunks, record id to delete if a stream is in dedup mode to avoid stale documents in the vector store)
+        """
+        doc = self._generate_document(record)
+        if doc is None:
+            self.logger.warning(f"Record {str(record.data)[:250]}... does not contain any text fields. Skipping.")
+            return [], None
+        chunks = self._split_document(doc)
+        id_to_delete = doc.metadata[METADATA_RECORD_ID_FIELD] if METADATA_RECORD_ID_FIELD in doc.metadata else None
+        return chunks, id_to_delete
+
+    def _generate_document(self, record: AirbyteRecordMessage) -> Optional[Document]:
+        relevant_fields = self._extract_relevant_fields(record)
+        if len(relevant_fields) == 0:
+            return None
+        metadata = self._extract_metadata(record)
+        text = stringify_dict(relevant_fields)
+        return Document(page_content=text, metadata=metadata)
+
+    def _extract_relevant_fields(self, record: AirbyteRecordMessage) -> dict:
+        relevant_fields = {}
+        if self.text_fields:
+            for field in self.text_fields:
+                values = dpath.util.values(record.data, field, separator=".")
+                if values and len(values) > 0:
+                    relevant_fields[field] = values
+        else:
+            relevant_fields = record.data
+        return relevant_fields
+
+    def _extract_metadata(self, record: AirbyteRecordMessage) -> dict:
+        metadata = {}
+        if self.metadata_fields:
+            for field in self.metadata_fields:
+                try:
+                    value = dpath.util.get(record.data, field, separator=".")
+                    dpath.util.new(metadata, field, value, separator=".")
```

Comment:
I don't think this will properly work if there are wildcards in the name:
```
{
  a: { x: "abc" }
}
```

If the field is specified as `*.x`, it won't map it back to that shape, but instead to `{'*': {'x': 'abc'}}`

Also, it will fail with a rather cryptic error message if there are multiple leaves in the object matching the specified path.

I think the easiest approach here is to not allow wildcards in the field name and document that in the description of the field and the markdown file.